File: CodingSamples/Fundamentals/Inheritance/ClassTypeTest/DemoApp1/Payroll/Employee.cs
namespace Payroll
{
    //user-defined reference type
    class Employee
    {
        private int hours;
        private float rate;

        //parameterized constructor
        public Employee(int h, float r)
        {
            hours = h;
            rate = r;
        }

        //parameterless constructor which calls the parameterized constructor
        public Employee() : this(0, 50)
        {
        }

        //Property - is a member of a class that enables getting/setting 
        //a value associated with its instance using field-access syntax
        public int Workdays
        {
            get
            {
                return hours / 8;
            }
            set
            {
                hours = 8 * value;
            }
        }

        public float DailyWages
        {
            get { return 8 * rate; }
            set { rate = value / 8; }
        }

        //a method declared with virtual keyword is overridable
        //in the derived class
        public virtual double Income()
        {
            double payment = hours * rate;
            int ot = hours - 180;
            if(ot > 0)
                payment += 50 * ot;
            return payment;

        }
    }
}

File: CodingSamples/Java/Fundamentals/Fundamentals/Inheritance/SubclassTest/demo1/payroll/Employee.java
package payroll;

//a class defined with public keyword is visible
//to classes defined outside of its package
public class Employee {
    
    private int hours;
    private float rate;

    public Employee(int h, float r) {
        hours = h;
        rate = r;
    }

    //parameterless constructor
    public Employee() {
        this(0, 50);
    }

    //defining a logical Workdays property using get/set methods
    public int getWorkDays() {
        return hours / 8;
    }

    public void setWorkDays(int value) {
        hours = 8 * value;
    }

    public float getDailyWages() {
        return 8 * rate;
    }

    public void setDailyWages(float value) {
        rate = value / 8;
    }

    public double income() {
        double payment = hours * rate;
        int ot = hours - 180;
        if(ot > 0)
            payment += ot * 50;
        return payment;
    }
}
